spring-cloud-sleuth-autoconfigure/src/test/java/org/springframework/cloud/sleuth/autoconfig/wavefront/WavefrontTracingIntegrationTests.java

```diff
@@ -0,0 +1,392 @@
+		// Less than a millis should round up to 1, but the test could take longer than
+		// 1ms
+		assertThat(spanRecord.durationMillis).isPositive();
+
+		// http
+		assertThat(spanRecord.tags).containsExactlyInAnyOrder(Pair.of("application", "IntegratedTracingTests"),
+				Pair.of("service", "test_service"), Pair.of("cluster", "none"), Pair.of("shard", "none"),
+				Pair.of("http.method", "GET"), Pair.of("http.path", "/api/fn/10"),
+				Pair.of("mvc.controller.class", "WebMvcController"), Pair.of("mvc.controller.method", "fn"),
+				Pair.of("span.kind", "server"), Pair.of("ipv4", Platform.get().linkLocalIp()));
+	}
+
+	@Test
+	void http_badRequest_setsStatusCodeAndErrorTrueTags() {
+		this.client.get().uri("/badrequest").exchange().expectStatus().isBadRequest();
+
+		SpanRecord spanRecord = takeRecord(spanRecordQueue);
+
+		// http
+		assertThat(spanRecord.tags).contains(Pair.of("http.status_code", "400"), Pair.of("error", "true"));
+	}
+
+	@Test
+	void setsStatusCodeAndErrorTrueTags_opentracing() {
+		this.client.get().uri("/error/opentracing").exchange().expectStatus().is5xxServerError();
+
+		SpanRecord spanRecord = takeRecord(spanRecordQueue);
+		// http
+		assertThat(spanRecord.tags).contains(Pair.of("http.status_code", "500"), Pair.of("error", "true") // retains
+																											// the
+																											// boolean
+																											// true
+		);
+	}
+
+	@Test
+	void setsStatusCodeAndErrorTrueTags_brave() {
+		this.client.get().uri("/error/brave").exchange().expectStatus().is5xxServerError();
+
+		SpanRecord spanRecord = takeRecord(spanRecordQueue);
+		// http
+		assertThat(spanRecord.tags).contains(Pair.of("http.status_code", "500"), Pair.of("error", "true") // deletes
+																											// the
+																											// user
+																											// message
+		);
+	}
+
+	@Test
+	void setsStatusCodeAndErrorTrueTags_exception() {
+		this.client.get().uri("/error/exception").exchange().expectStatus().is5xxServerError();
+
+		// http
+		assertThat(takeRecord(spanRecordQueue).tags).contains(Pair.of("http.status_code", "500"),
+				Pair.of("error", "true") // deletes the exception message
+		);
+	}
+
+	@Test
+	void setsStatusCodeAndErrorTrueTags_thrownException() {
+		this.client.get().uri("/throws").exchange().expectStatus().is5xxServerError();
+
+		SpanRecord spanRecord = takeRecord(spanRecordQueue);
+		// http
+		assertThat(spanRecord.tags).contains(
+				/* Pair.of("http.status_code", "500"), */ // Able to return error=true
+															// span tag but not
+															// http.status_code span tag
+				Pair.of("error", "true") // deletes the exception message
+		);
+	}
+
+	/** Helps ensure test bugs don't result in hung tests! */
+	<R> R takeRecord(BlockingDeque<R> queue) {
+		R result;
+		try {
+			result = queue.poll(3, TimeUnit.SECONDS);
+		}
+		catch (InterruptedException e) {
+			Thread.currentThread().interrupt();
```

Comment:
Wasn't it already interrupted (and the interrupt status is cleared)? Why does this needed?

Reply:
That's a copy from the previous solution.